### ISSUE TITLE
[CT-1519] [Bug] Better error message for misspecified dist key yaml config

### ISSUE THREAD

**Issue by dluftspring** [dbt-labs/dbt-redshift#224]
### Is this a new bug in dbt-redshift?

- [X] I believe this is a new bug in dbt-redshift
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

If you set the dist key in a `model.yml` file like

```yaml
version: 2
models:
  - name: a_real_table
    config:
      sort:
        - primary_key
        - foreign_key
      dist:
        - primary_key
```

dbt will throw the following error

`list has no attribute strip`

Which is generic and non descriptive of what the user should fix. The issue is that since dist keys can only be single valud properties it should be specified as `dist: primary_key`.

### Expected Behavior

A better message that isn't exposing the underlying python machinery and indicates what the user should do to fix the issue e.g. Single valued properties cannot be specified as a list

### Steps To Reproduce

See Current Behaviour. If you set up a model like that in any dbt project you will reproduce the error

### Environment

```markdown
- OS: Mac
- Python: 3.9
- dbt-core:
- dbt-redshift: 1.3.0
```

**Comment by dbeatty10** [comment]
@dluftspring thank you for noticing and reporting this!

You are spot-on with your assessment, and it definitely makes sense to proactively notice where the type of input is unexpected and indicate how to fix it.

I'm going to label this as a "good_first_issue" for a contributor to pick up.

### Implementation suggestion
[Raise an error](https://docs.getdbt.com/reference/dbt-jinja-functions/exceptions) at the beginning of this macro if `dist` is a list instead of a string:
https://github.com/dbt-labs/dbt-redshift/blob/717018aec33e3a04a74445cff2637bd0629ff196/dbt/include/redshift/macros/adapters.sql#L2-L14

**Comment by dluftspring** [comment]
If that's all that needs to be done I can contribute the fix for this

**Comment by dbeatty10** [comment]
Awesome @dluftspring !

When you open up a PR for this, just add the following as the first line:
```
resolves #224
```

This will help the reviewer of the PR link back to this conversation.
